### PR TITLE
vk-cli: init at 0.7.6

### PR DESCRIPTION
**Pull request by dan4ik605743**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
I'm not sure what the policy is for adding abandoned projects to nixpkgs but this might be better added to your own [NUR](https://github.com/nix-community/nur)

**Comment by dan4ik605743** [comment]
> I'm not sure what the policy is for adding abandoned projects to nixpkgs but this might be better added to your own [NUR](https://github.com/nix-community/nur)

I already have it in NUR. But I thought why not in nixpkgs. The package is working.

**Comment by cole-h** [comment]
@ofborg eval

(Sorry, ofborg's certs recently had an issue, meaning that it was unable to
react to PRs. This has been fixed, but all PRs filed in this period will need
ofborg to be manually kicked off.)

**Comment by dan4ik605743** [comment]
> @ofborg eval
> 
> (Sorry, ofborg's certs recently had an issue, meaning that it was unable to
> react to PRs. This has been fixed, but all PRs filed in this period will need
> ofborg to be manually kicked off.)

Well thank you

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    url = "https://github.com/vk-cli/vk/releases/download/${version}/vk-0.7.6-64-bin.7z";
+    sha256 = "sha256-Y40oLjddunrd7ZF1JbCcgjSCn8jFTubq69jhAVxInXw=";
+  };
+
+  nativeBuildInputs = [
+    p7zip
+  ];
+
+  buildInputs = [
+    curlFull
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:26]: Why not just curl?

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      curlFull
+      zulip
+      glibc
+    ];
+  in ''
+    patchelf \
+    --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+    --set-rpath "${libPath}" \
+    $out/bin/vk-cli
+    '';
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:53]:
```suggestion
  postFixup = ''
    patchelf $out/bin/vk-cli \
    --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
    --set-rpath "${lib.makeLibraryPath [ curlFull zulip glibc ]}"
    '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  installPhase = ''
+    mkdir -p $out/bin/
+    mv $TMP/vk-0.7.6-64-bin vk-cli
+    install -D vk-cli --target-directory=$out/bin/
+  '';
+
+  postFixup = ''
+    patchelf $out/bin/vk-cli \
+    --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+    --set-rpath "${lib.makeLibraryPath [ curl zulip glibc ]}"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:45]:
```suggestion
    patchelf $out/bin/vk-cli \
      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
      --set-rpath "${lib.makeLibraryPath [ curl zulip glibc ]}"
```

**Comment by 06kellyjac** [comment]
you'll want to squash the 3 commits into one :)

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+, ncurses
+, openssl
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vk-cli";
+  version = "0.7.6";
+
+  src = fetchurl {
+    url = "https://github.com/vk-cli/vk/releases/download/${version}/vk-0.7.6-64-bin.7z";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:17]:
```suggestion
    url = "https://github.com/vk-cli/vk/releases/download/${version}/vk-${version}-64-bin.7z";
```

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  ];
+
+  unpackPhase = ''
+    mkdir -p $TMP/
+    7z x $src -o$TMP/
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin/
+    mv $TMP/vk-0.7.6-64-bin vk-cli
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:38]:
```suggestion
    mv $TMP/vk-${version}-64-bin vk-cli
```

**Comment by dan4ik605743** [comment]
@SuperSandro2000 Okay, I'll fix it later. please check the rest of the PR.

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix (hunk 5 above)
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:38]: You are sure $TMP stays the same here?

Comment (dan4ik605743) [reply pkgs/applications/networking/instant-messengers/vk-cli/default.nix:38]: > You are sure $TMP stays the same here?

I tested this bill. And VK-CLI itself. That's right.

**Comment by dan4ik605743** [comment]
@SuperSandro2000 ?

**Comment by dan4ik605743** [comment]
@SuperSandro2000 Please check my PR

File: pkgs/applications/networking/instant-messengers/vk-cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  installPhase = ''
+    mkdir -p $out/bin/
+    mv $TMP/vk-${version}-64-bin vk-cli
+    install -D vk-cli --target-directory=$out/bin/
+  '';
+
+  postFixup = ''
+    patchelf $out/bin/vk-cli \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${lib.makeLibraryPath [ curl zulip glibc ]}"
```
Comment (andersk) [inline pkgs/applications/networking/instant-messengers/vk-cli/default.nix:45]: What does this depend on `zulip` for? `zulip` is an application; it does not have any libraries.

#132052

**Pull request merged**